### PR TITLE
Working on ILiquidTemplateEventHandler.

### LINKED ISSUE #979: Provide root objects in liquid from services
Create and extensible way to provide root properties in a template context.
To be able to use `Queries.RecentBlogPosts` for instance. Could be used also for `User`, `Site` which would then be provided by their respective modules.

### PR DESCRIPTION
**Pull request by jtkech**
Fixes #979 

- Here we still resolve `Filters` in `LiquidViewTemplate` but no more in `LiquidTemplateManager`.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Users/Liquid/UserLiquidValueProvider.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+        public UserLiquidValueProvider(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public Task PopulateValuesAsync(IDictionary<string, object> values)
+        {
+            values.Add("User", _httpContextAccessor.HttpContext.User);
+            values.Add("User.Identity", _httpContextAccessor.HttpContext.User.Identity);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Users/Liquid/UserLiquidValueProvider.cs]: Shouldn't be necessary as it's exposed for the `ClaimsPrincipal` type

File: src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs [hunk 2]
```diff
@@ -24,8 +23,6 @@ public class Startup : StartupBase
         static Startup()
         {
             TemplateContext.GlobalMemberAccessStrategy.Register<ContentItem>();
-            TemplateContext.GlobalMemberAccessStrategy.Register<ClaimsPrincipal>();
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs]: Why did you remove it, it's necessary to do `User.Identity.Name`

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs [hunk 3]
```diff
@@ -184,21 +192,33 @@ public static void Contextualize(this TemplateContext context, RazorPage page, o
...
+            }
 
-            // TODO: Extract the request culture instead of the default site's one
-            if (site.Culture != null)
+            foreach (var value in values)
             {
-                context.CultureInfo = new CultureInfo(site.Culture);
+                var type = value.Value.GetType();
+
+                if (Type.GetTypeCode(type) == TypeCode.Object)
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs]: Meh, why not expose the context directly in the `ILiquidValueProvider` instead? Then we could just extend anything, and add the type if necessary. Then the module can decide if the type should be registered globally or on the context.

File: src/OrchardCore/OrchardCore.Liquid.Abstractions/ILiquidValueProvider.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Liquid
+{
+    public interface ILiquidValueProvider
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Liquid.Abstractions/ILiquidValueProvider.cs]: If you agree with the change above, then we can just call it `ILiquidTemplateEventHandler` with a `RenderingAsync` method taking a `TemplateContext`

File: src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs (hunk 2 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs]: Ok. I thought this would need to be done through the `O.Users` module

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs (hunk 3 above)
Comment (jtkech) [reply src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs]: Yes, simpler, i was not sure we can allow this.

> Then the module can decide if the type should be registered globally or on the context

You also say globally, do you mean e.g that `O.Users` could register `ClaimsPrincipal` globally? Here or in a static constructor? Anyway, when registering globally, as i remember, we would need to consider concurrent writing.

File: src/OrchardCore/OrchardCore.Liquid.Abstractions/ILiquidValueProvider.cs (hunk 4 above)
Comment (jtkech) [reply src/OrchardCore/OrchardCore.Liquid.Abstractions/ILiquidValueProvider.cs]: Ok, i will do it.

**Comment by jtkech** [comment]
So, i did the changes.

- Right now, `ClaimsPrincipal` and `ClaimsIdentity` are registered in the user handler but not globally.

- We could do it globally but we would have to consider concurent writings. Maybe better to do it in the static startup of `O.Users` but not sure it is better if several static startup write in global strategies.

- Finally, let me know if you prefer to register these types globally and in the static startup of `O.Liquid`.

**Comment by sebastienros** [comment]
Both are fine, so I won't say anything about having them in the context, unless we prove it has some perf impact, which it shouldn't.

File: src/OrchardCore.Modules/OrchardCore.Users/Liquid/UserLiquidValueProvider.cs (hunk 1 above)
Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Users/Liquid/UserLiquidValueProvider.cs]: You forgot to remove this.
